src/components/toolbar/mdc-toolbar.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <header class="mdc-toolbar">
+    <!-- here are some sugar slots -->
+    <div class="mdc-toolbar__row" v-if="$slots.icon || $slots.title" ref="firstRow">
+      <section class="mdc-toolbar__section mdc-toolbar__section--align-start" v-if="$slots.icon">
+        <slot name="icon" />
+      </section>
+      <section class="mdc-toolbar__section mdc-toolbar__section--align-start" v-if="$slots.title">
+         <slot name="title" />
+      </section>
+    </div>
+    <!-- if the user doesn't want to use the sugar slots then they can inject their own content here -->
+    <slot/>
+  </header>
+</template>
+
+<script lang="babel">
+  import MDCToolbarFoundation from '@material/toolbar/foundation'
+  import * as util from '@material/toolbar/util'
+
+  export default {
+    name: 'Toolbar',
+    props: ['titleElement', 'flexibleRowElement', 'fixedAdjustElement', 'firstRowElement'],
```

Comment:
You don't, that's what all the `if (el)` checks are for in the adapter. The alternative to using props is either forcing the user to use slots (we shouldn't do that) or do query selector on the DOM (shouldn't do that either).